golang/src/main/java/com/regnosys/rosetta/generator/golang/object/GolangModelObjectGenerator.xtend

```diff
@@ -0,0 +1,130 @@
+package com.regnosys.rosetta.generator.golang.object
+
+import com.google.inject.Inject
+import com.regnosys.rosetta.RosettaExtensions
+import com.regnosys.rosetta.rosetta.RosettaClass
+import com.regnosys.rosetta.rosetta.RosettaMetaType
+import java.util.List
+
+import static com.regnosys.rosetta.generator.golang.util.GolangModelGeneratorUtil.*
+
+import static extension com.regnosys.rosetta.generator.util.RosettaAttributeExtensions.*
+import java.util.Map
+import java.util.HashMap
+import com.regnosys.rosetta.rosetta.simple.Data
+import com.regnosys.rosetta.generator.object.ExpandedAttribute
+import java.util.Set
+import com.google.common.collect.Lists
+
+class GolangModelObjectGenerator {
+
+	@Inject extension RosettaExtensions
+	@Inject extension GolangModelObjectBoilerPlate
+	@Inject extension GolangMetaFieldGenerator
+	
+	static final String CLASSES_FILENAME = 'types.go'
+	static final String META_FILENAME = 'metatypes.go'
+	
+	def Map<String, ? extends CharSequence> generate(Iterable<Data> rosettaClasses, Iterable<RosettaMetaType> metaTypes, String version) {
+		val result = new HashMap		
+		val enumImports = rosettaClasses
+				.map[allExpandedAttributes].flatten
+				.map[type]
+				.filter[isEnumeration]
+				.map[name]
+				.toSet
+		
+		val classes = rosettaClasses.sortBy[name].generateClasses(enumImports, version).replaceTabsWithSpaces
+		result.put(CLASSES_FILENAME, classes)
+		val metaFields = generateMetaFields(metaTypes, version).replaceTabsWithSpaces
+		result.put(META_FILENAME, metaFields)
+		result;
+	}
+	
+	def Map<String, ? extends CharSequence> generate2(Iterable<Data> rosettaClasses, Iterable<RosettaMetaType> metaTypes, String version) {
+		val result = new HashMap		
+		val enumImports = rosettaClasses
+				.map[allExpandedAttributes].flatten
+				.map[type]
+				.filter[isEnumeration]
+				.map[name]
+				.toSet
+		
+		val classes = rosettaClasses.sortBy[name].generateClasses2(enumImports, version).replaceTabsWithSpaces
+		result.put(CLASSES_FILENAME, classes)
+		val metaFields = generateMetaFields(metaTypes, version).replaceTabsWithSpaces
+		result.put(META_FILENAME, metaFields)
+		result;
+	}
+
+	private def generateClasses(List<Data> rosettaClasses, Set<String> importedEnums,  String version) {
+	'''	
+	package types
+	
+	«fileComment(version)»	
+	
+	import . "cdm/metatypes";
+	import . "cdm/enums";
+	
```

Comment:
enums should be all under `org.isda.cdm` whereas metafields should be under `org.isda.cdm.metafields`

Reply:
Cannot have dots  -- we are constrained by Golang here (its package system is very different from the Java's). Names "metattypes", "enums", "types" are taken from file names in the typescript and scala generators.